ambari-logsearch/ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/input/InputSocket.java

```diff
@@ -0,0 +1,165 @@
+
+  private ServerSocket serverSocket;
+  private Thread thread;
+  private int port;
+  private String protocol;
+  private boolean secure;
+  private boolean log4j;
+
+  @Override
+  public void init(LogFeederProps logFeederProperties) throws Exception {
+    super.init(logFeederProperties);
+    port = (int) ObjectUtils.defaultIfNull(((InputSocketDescriptor)getInputDescriptor()).getPort(), 0);
+    if (port == 0) {
+      throw new IllegalArgumentException(String.format("Port needs to be set for socket input (type: %s)", getInputDescriptor().getType()));
+    }
+    protocol = (String) ObjectUtils.defaultIfNull(((InputSocketDescriptor)getInputDescriptor()).getProtocol(), "tcp");
+    secure = (boolean) ObjectUtils.defaultIfNull(((InputSocketDescriptor)getInputDescriptor()).isSecure(), false);
+    log4j = (boolean) ObjectUtils.defaultIfNull(((InputSocketDescriptor)getInputDescriptor()).isLog4j(), false);
+  }
+
+  @Override
+  public boolean monitor() {
+    if (isReady()) {
+      LOG.info("Start monitoring socket thread...");
+      thread = new Thread(this, getNameForThread());
+      thread.start();
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public void start() throws Exception {
+    LOG.info("Starting socket server (port: {}, protocol: {}, secure: {})", port, protocol, secure);
+    ServerSocketFactory socketFactory = secure ? SSLServerSocketFactory.getDefault() : ServerSocketFactory.getDefault();
+    InputSocketMarker inputSocketMarker = new InputSocketMarker(this, port, protocol, secure, log4j);
+    LogsearchConversion loggerConverter = new LogsearchConversion();
+    try {
+      serverSocket = socketFactory.createServerSocket(port);
+      while (!isDrain()) {
+        Socket socket = serverSocket.accept();
+        if (log4j) {
+          try (ObjectInputStream ois = new ObjectInputStream(new BufferedInputStream(socket.getInputStream()))) {
+            LoggingEvent loggingEvent = (LoggingEvent) ois.readObject();
+            String jsonStr = loggerConverter.createOutput(loggingEvent);
+            LOG.trace("Incoming socket logging event: " + jsonStr);
+            outputLine(jsonStr, inputSocketMarker);
+          }
+        } else {
+          try (BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));) {
+            String line = in.readLine();
+            LOG.trace("Incoming socket message: " + line);
+            outputLine(line, inputSocketMarker);
+          }
+        }
+      }
+    } catch (SocketException socketEx) {
+      LOG.warn("{}", socketEx.getMessage());
+    } finally {
+      serverSocket.close();
+    }
+  }
+
+  @Override
+  public void setDrain(boolean drain) {
+    super.setDrain(drain);
+    LOG.info("Stopping socket input: {}", getShortDescription());
+    try {
+      serverSocket.close();
+      setClosed(true);
+    } catch (Exception e) {
+      LOG.error("Error during closing socket input: {}", e);
```

Comment:
The stacktrace is lost because Exception.toString() will be invoked.

Reply:
@kasakrisz i think it wont act as a string parameter more like a theowable parameter so it eont be the case, but of course that would mean {} is not requird